### PR TITLE
Change "and" to allow "short", use term

### LINKED ISSUE #137: Make "and" attribute in name lists more configurable
The current definition of the "and" attribute in the specification is:

> Specifies the delimiter between the second to last and last name of the names in a name variable. Allowed values are “text” (selects the “and” term, e.g. “Doe, Johnson and Smith”) and “symbol” (selects the ampersand, e.g. “Doe, Johnson & Smith”).

So there are only two options: the "long" form of the "and" term or the hardcoded ampersand. There have been several reports that this is too limited. Some languages don't seem to use ampersands at all (we received some private communication from @cpina that this is e.g. the case for Spanish and Catalan: '"y" (and) and in Catalan they do like in Spanish ("i", it's "and" in Catalan).'), and some styles require different text forms between the in-text citations and bibliography, e.g. an abbreviated and unabbreviated form (see https://github.com/citation-style-language/styles/pull/2571#issuecomment-287253895).

Most of this could be solved by changing the spec to have `and="symbol"` select the "symbol" form of the "and" term, and by defining the ampersand as translation for this term form in the locale files, which is a relatively easy fix.

A larger but more flexible change would be to make it possible to also e.g. choose the "short" form of the "and" term. This would be possible by e.g. having the "and" attribute directly select the desired term-form, with e.g. attribute values "long", "short", and "symbol" (`and="long"` would replace `and="text"`). Might also want to rename the attribute to "and-form", although I don't think that's very pretty.

### PR DESCRIPTION
**Pull request by bdarcus**
`and="text"` -> `and="long"`; makes clear rendering for "and" should be based on term definition, and so also to allow configuration of `and="symbol"`.

closes #137

### CODE REVIEW

**Comment by bdarcus** [comment]
Note: I used the Atom editor here, which removed a bunch of whitespace characters that didn't need to be there.

File: csl.rnc [hunk 1]
```diff
@@ -430,174 +430,178 @@ div {
...
+
       ## Use the "and" term (e.g., "Doe, Johnson and Smith").
       "text"
-      | 
-        ## Use the "ampersand" (e.g., "Doe, Johnson & Smith").
+      |
+        ## Use the "ampersand" by default, or whatever is defined in the term (e.g., "Doe, Johnson & Smith").
         "symbol"
+      |
+        ## Use a "short" for as defined in the term. Not common in English, but is seen in other languages.
```
Comment (fbennett) [inline csl.rnc:477]: for -> form

Comment (bdarcus) [reply csl.rnc]: Doh! Thanks; fixed.

**Comment by bdarcus** [comment]
> My inclination is to simplify this explanation. Replace `"text"` with `"long"`.

What would be the style compatibility implications of that though? This was why I didn't make that change.

PS - this is the kind of comment that ideally is in the "review" mode.

**Comment by bdarcus** [comment]
Except, maybe because frank did a review you  don't have that option?

**Comment by denismaier** [comment]
> 
> 
> Except, maybe because frank did a review you don't have that option?

No, should be possible. I could request a review from @bwiernik (or myself, for that matter), so bwiernik might be able to do so as well.

**Comment by bdarcus** [comment]
OK,  but if I request reviews from the "PR reviewers" team, I
basically want whoever has the time to review. So if it were to close
when one review, that would be less than ideal.

**Comment by bdarcus** [comment]
> With that in mind, I think replacing `and="text"` with `and="long"` is the best state to reach. At a minimum, we should add `and="long"` as the default and describe `and="text"` as a deprecated alias of `and="long"`.
> 
> Doing a batch replace of `and="text"` with `and="long"` in the official styles is simple enough. If we want to maintain compatibility for user styles in the wild, then let's add `and="long"` now and deprecate `and="text"`, then remove `and="text"` in CSL 1.1.

OK, gotcha.

So this is the key strategic question. Anyone else want to weigh in here?

**Comment by denismaier** [comment]
So, what exactly is the current suggestion?

**Comment by bdarcus** [comment]
> So, what exactly is the current suggestion?

Really two options:

1. deprecate `and="text"` now and replace with `and="long"`
2. add  `and="long"`now and deprecate `and="text"` later (say 1.1)

**Comment by denismaier** [comment]
> * deprecate `and="text"` now and replace with `and="long"`
That would entail a schema update to 1.1?

> * add  `and="long"`now and deprecate `and="text"` later (say 1.1)
This would be possible with a 1.0.x minor release?

Am I right here?

And we will also have a third option `symbol` alongside `long` and `short`? You weren't talking about dropping this as well?

**Comment by bdarcus** [comment]
> This would be possible with a 1.0.x minor release?

I think so.

I'd rather just do 1 for 1.1. It's cleaner and less confusing.

> And we will also have a third option symbol alongside long and short? 

Yes.

**Comment by bdarcus** [comment]
I've updated the code and doc strings to where I think it needs to be, for 1.1 release.

**Comment by bwiernik** [comment]
To be clear on my word usage here: “Deprecate” means “still around, but don’t use. We are warning you that it will go away in the future”. “Remove” means actually removing it.

**Comment by denismaier** [comment]
> To be clear on my word usage here: “Deprecate” means “still around, but don’t use. We are warning you that it will go away in the future”. “Remove” means actually removing it.

Would also be my understanding.

**Comment by denismaier** [comment]
I have no general objections to adding this with a 1.1. release. But I just hope that will be before 2025.

**Comment by bdarcus** [comment]
:-)

I hope we can do it this summer, accepting that we won't have the spec
stuff that Norm is looking for done at that point.

**Comment by bdarcus** [comment]
I'm hoping to merge this to the v1.1 branch later on Friday, so if any input (including affirmative, that result in an "approve" review), please get it in by then.

**Comment by bdarcus** [comment]
> My inclination is to simplify this explanation. Replace `"text"` with `"long"`. Describe in the spec that the localized term `and` is rendered, with `form` given by the `and` attribute on `names`.

I missed this earlier. I can still update the language, as this is a bit more precise.

**Pull request merged**